### PR TITLE
Adding document talking about tests for this repo

### LINKED ISSUE #62: Document how our automated testing in openshift-ci works
Document how our automated testing in openshift-ci works.
Topics to be covered:

1)  How to run our tests locally and how our local runs differ from the openshift-ci runs
2)  The contents of our tests....how you can write your own
3)  How to get your repo added to openshift-ci
4)  What our openshift-ci configuration looks like...and why

### PR DESCRIPTION
**Pull request by crobby**
Closes #62

### CODE REVIEW

File: tests/TESTING.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+Each test script has a small amount of boilerplate code followed by a series of bash tests.  Each test could call out to another utility/language/whatever.  The utilities available in the testing library can check for specific results in text/exit code/etc of each call.  Any test lines that produce a failed result are tabulated and reported at the end of the test runs.  Of course, the stdout/stderr of each failed call is also available in addition to whatever other logging your test call might produce.  Here’s what I would call the main building block of the tests:  https://github.com/openshift/origin/blob/master/hack/lib/cmd.sh It defines what amount to wrappers around whatever calls you want to make in your tests and handles the parsing of the result text/exit codes.
+
+## Integration with OpenShift-ci:
```
Comment (vpavlin) [inline tests/TESTING.md:9]: maybe change to "OpenShift CI"?

File: tests/TESTING.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+The first step toward integrating with OpenShift CI is granting access to the OpenShift CI Robot and the OpenShift Merge Robot entities in the settings of the repo.  Once that is complete, you can contact the openshift-ci team and they will set up the necessary webhooks in the target repo.
```
Comment (vpavlin) [inline tests/TESTING.md:11]: Can you add any links for docs?

File: tests/TESTING.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+Next is the prow configuration.  The configuration files are kept in the https://github.com/openshift/release repository.  Under the “core_services/prow” directory, you’ll need to modify _config.yaml and _plugins.yaml in order to have your repository included in the configuration.  Submit a PR to that repo and when it merges, you’re all set.  As for the contents of your changes, unless you know exactly what you want, it might be useful to start by adding your repo with settings copied from another repository already in the config.
```
Comment (vpavlin) [inline tests/TESTING.md:13]: Make `core_services/prow` a code formatted and ideally any other file reference in the text

File: tests/TESTING.md (hunk 2 above)
Comment (crobby) [reply tests/TESTING.md:11]: I can link to the repos for the tools, but don't want to include any "internal only" docs

Comment (vpavlin) [reply tests/TESTING.md]: Yeah, if there is nothing else public, then we are good

**Pull request merged**